shell/platform/darwin/ios/framework/Source/FlutterStandardCodec.mm

```diff
@@ -242,24 +279,37 @@ - (void)writeValue:(id)value {
   } else if ([value isKindOfClass:[NSNumber class]]) {
     NSNumber* number = value;
     const char* type = [number objCType];
-    if (strcmp(type, @encode(BOOL)) == 0) {
+    if ([self isBool:number type:type]) {
       BOOL b = number.boolValue;
       [self
           writeByte:(b ? FlutterStandardFieldTrue : FlutterStandardFieldFalse)];
-    } else if (strcmp(type, @encode(int)) == 0) {
+    } else if (strcmp(type, @encode(signed int)) == 0 ||
+               strcmp(type, @encode(signed short)) == 0 ||
+               strcmp(type, @encode(unsigned short)) == 0 ||
+               strcmp(type, @encode(signed char)) == 0 ||
+               strcmp(type, @encode(unsigned char)) == 0) {
       SInt32 n = number.intValue;
```

Comment:
Consider introducing a switch here, for performance. That probably implies using constant char values instead of the symbolic `@encode(some type)`